cmd/parquetgen/gen/template_optional.go

```diff
@@ -26,9 +26,13 @@ func (f *{{.FieldType}}) Schema() parquet.Field {
 }
 
 func (f *{{.FieldType}}) Write(w io.Writer, meta *parquet.Metadata) error {
-	var buf bytes.Buffer
+	buf := buffpool.Get()
```

Comment:
why do we need a buffer pool here? Can we just write our `bs` from stack or in that case it will escape?

Reply:
I tried the following implementation:
```
func (f *Float64Field) Write(w io.Writer, meta *parquet.Metadata) error {
	bs := make([]byte, 8 * len(f.vals))
	for i, v := range f.vals {
		binary.LittleEndian.PutUint64(bs[i*8:(i+1)*8], math.Float64bits(v))
	}
	return f.DoWrite(w, meta, bs, len(f.vals), f.stats)
}
```
it was indeed a bit faster, but also required a bit more memory:
```
// allocate full slice
BenchmarkWrite/opt-8      24   464.75 ms/op   213750980 B/op   4313175 allocs/op
// with buffer
BenchmarkWrite/opt-8      25   471.62 ms/op   202659443 B/op   4312402 allocs/op
```
so I would prefer to keep the current implementation